WeCantSpell.Hunspell/WordList.Query.cs

```diff
@@ -2014,48 +1886,48 @@ private WordEntry CheckWordSuffix(Affix<SuffixEntry> affix, string word, AffixEn
 
                 // if all conditions are met then check if resulting
                 // root word in the dictionary
-                if (entry.Conditions.IsEndingMatch(tmpstring))
+                if (affix.Conditions.IsEndingMatch(tmpSpan))
                 {
-                    var aFlag = affix.AFlag;
-                    foreach (var heDetail in LookupDetails(tmpstring))
+                    if (TryLookupDetails(tmpSpan, out var tmpString, out var details))
                     {
-                        if (
-                            (
-                                heDetail.ContainsFlag(aFlag)
-                                ||
+                        foreach (var heDetail in details)
+                        {
+                            if (
                                 (
-                                    pfx is not null
-                                    &&
-                                    pfx.Entry.ContainsContClass(aFlag)
+                                    heDetail.ContainsFlag(affix.AFlag)
+                                    ||
+                                    (
+                                        pfx?.ContainsContClass(affix.AFlag) == true
+                                    )
                                 )
-                            )
-                            &&
-                            (
-                                !optFlagsHasCrossProduct
-                                ||
+                                &&
                                 (
```

Comment:
This should be impossible to pass if `optFlagsHasCrossProduct` is true and `pfx` is null . Could save a lot of looping!

Reply:
This should already be handled.